### PR TITLE
Allow a callback to be passed to cleanDeletedBlocks making it interruptible.

### PR DESCRIPTION
**Pull request by nick4598**
imodel-native: https://github.com/iTwin/imodel-native/pull/787

### CODE REVIEW

File: core/backend/src/CloudSqlite.ts [hunk 1]
```diff
@@ -360,8 +360,23 @@ export namespace CloudSqlite {
      * a 404 error. Default is 1 hour.
      */
     nSeconds?: number;
-    /** if enabled, outputs verbose logs about the cleanup process. These would include outputting blocks which are determined as eligible for deletion. */
+    /** if enabled, outputs verbose logs about the cleanup process. These would include outputting blocks which are determined as eligible for deletion.
```
Comment (ben-polinsky) [inline core/backend/src/CloudSqlite.ts:363]:
```suggestion
    /** if enabled, outputs verbose logs about the cleanup process. Output includes blocks determined eligible for deletion.
```

File: core/backend/src/CloudSqlite.ts [hunk 2]
```diff
@@ -360,8 +360,23 @@ export namespace CloudSqlite {
...
+     * @default false
+    */
     debugLogging?: boolean;
+    /** If true, iterates over all blobs in the cloud container to add blocks that are 'orphaned' to the delete list in the manifest.
+     * Orphaned blocks are created when a client abruptly halts, is disconnected or encounters an error while uploading a change.
```
Comment (ben-polinsky) [inline core/backend/src/CloudSqlite.ts:368]:
```suggestion
     * Orphaned blocks are created when a client abruptly halts, is disconnected, or encounters an error while uploading a change.
```
(oxford comma nit :))

File: core/backend/src/CloudSqlite.ts [hunk 3]
```diff
@@ -515,8 +521,8 @@ export namespace CloudSqlite {
      */
     copyDatabase(dbName: string, toAlias: string): Promise<void>;
 
-    /** Remove a database from this CloudContainer.
-     * @see cleanDeletedBlocks
+    /** Remove a database from this CloudContainer, moving all of its no longer used blocks to the delete list in the manifest.
```
Comment (ben-polinsky) [inline core/backend/src/CloudSqlite.ts:524]:
```suggestion
    /** Remove a database from this CloudContainer. Unused blocks are moved the delete list in the manifest.
```

File: core/backend/src/CloudSqlite.ts [hunk 4]
```diff
@@ -576,14 +582,52 @@ export namespace CloudSqlite {
     promise: Promise<boolean>;
   }
 
+  /**
+   * Clean any unused deleted blocks from cloud storage. When a database is written, a subset of its blocks are replaced
```
Comment (ben-polinsky) [inline core/backend/src/CloudSqlite.ts:586]: > When a database is written

"Written" seems like it means "created", is that the intent?
Written to? Updated?

**Review by ben-polinsky — CHANGES_REQUESTED** [review]

File: core/backend/src/CloudSqlite.ts (hunk 4 above)
Comment (nick4598) [reply core/backend/src/CloudSqlite.ts:586]: Hmm for a checkpoint, its created/written once. But I guess a database could be written to aka updated as well replacing teh subset of its blocks

Comment (nick4598) [reply core/backend/src/CloudSqlite.ts]: Changed it to `/**
   * Clean any unused deleted blocks from cloud storage. Unused deleted blocks can accumulate in cloud storage in a couple of ways:
   * 1) When a database is updated, a subset of its blocks are replaced by new versions, sometimes leaving the originals unused.
   * 2) A database is deleted with [[CloudContainer.deleteDatabase]]
   * In both cases, the blocks are not deleted immediately. Instead, they are scheduled for deletion at some later time.
   * Calling this method deletes all blocks in the cloud container for which the scheduled deletion time has passed.`

**Review by ben-polinsky — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Comment by grigasp** [comment]
Please try to complete this ASAP - the native part of the changes are merged and now all native PRs are failing validation.

**Comment by pmconne** [comment]
Integration:
```
  2 failing

  1) CloudSqlite
       cloud containers:
     AssertionError: expected promise to be rejected with an error including 'not authorized' but it was fulfilled with undefined
  

  2) CloudSqlite
       should be able to interrupt cleanDeletedBlocks operation:

      AssertionError: expected true to be false
      + expected - actual

      -true
      +false
      
      at Object.Sqlite.uploadFile (src/integration/AzuriteTest.ts:128:36)
      at Context.<anonymous> (src/integration/CloudSqlite.test.ts:467:20)
      at processImmediate (node:internal/timers:478:21)
```

Docs:
```
TypeDoc Reference Link Error: Reference to 'CleanDeletedBlocksOptions' in file reference/core-backend/sqlitedb/cloudsqlite/cloudsqlite.cleandeletedblocks/index.html was not found at File: core-backend/core/backend/src/CloudSqlite.ts Line: 594
One or more invalid links found.
```

**Comment by nick4598** [comment]
> Please try to complete this ASAP - the native part of the changes are merged and now all native PRs are failing validation.

Sorry about that, I definitely jumped the gun when I merged my native PR. Ive pushed up some fixes, so hopefully this pipeline run will pass.

**Pull request merged**